enumerable.rb

```diff
@@ -0,0 +1,145 @@
+# frozen_string_literal: true
+# rubocop:disable all
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+    i = 0
+      while i < size
+        yield self[i]
+        i += 1
+      end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+    i = 0
+      while i < size
+        yield(self[i], i)
+        i += 1
+      end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+    array2 = []
+      i = 0
+      while i < size
+        array2 << self[i] if yield self[i]
+        i += 1
+      end
+    array2
+  end
+
+  def my_all?(pattern = nil)
+    test = true
+    if(block_given?)
+      self.my_each do |x|
```

Comment:
```
%w[ant bear cat].all? { |word| word.length >= 3 } #=> true
%w[ant bear cat].all? { |word| word.length >= 4 } #=> false
%w[ant bear cat].all?(/t/)                        #=> false
[1, 2i, 3.14].all?(Numeric)                       #=> true
[nil, true, 99].all?                              #=> false
[].all?                                           #=> true
```

- [ ] my_all? when no block or argument is given returns true when none of the collection members are false or nil
- [ ] my_all? when a class is passed as an argument returns true if all of the collection is a member of such class
- [ ] my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern